sn_interface/src/network_knowledge/section_peers.rs

```diff
@@ -21,47 +19,71 @@ const ELDER_CHURN_EVENTS_TO_PRUNE_ARCHIVE: usize = 5;
 const ELDER_CHURN_EVENTS_TO_PRUNE_ARCHIVE: usize = 3;
 
 /// Container for storing information about (current and archived) members of our section.
-#[derive(Clone, Default, Debug)]
+#[derive(Clone, Default, Debug, Eq, PartialEq, Ord, PartialOrd)]
 pub(super) struct SectionPeers {
-    members: BTreeMap<XorName, SectionSigned<NodeState>>,
-    archive: BTreeMap<XorName, SectionSigned<NodeState>>,
+    members: BTreeMap<XorName, Decision<NodeState>>,
+    archive: BTreeMap<XorName, Decision<NodeState>>,
 }
 
 impl SectionPeers {
     /// Returns set of current members, i.e. those with state == `Joined`.
-    pub(super) fn members(&self) -> BTreeSet<SectionSigned<NodeState>> {
+    pub(super) fn members(&self) -> BTreeSet<NodeState> {
+        // TODO: ensure Decision has no Joined and Left combined within one
+        let mut node_state_list = BTreeSet::new();
+        for decision in self.members.values() {
+            for node_state in decision.proposals.keys() {
```

Comment:
I think these TODO's can be resolved in this PR with a filter.
```suggestion
            for node_state in decision.proposals.keys().filter(|node| matches!(node.state, MembershipState::Joined) {
```

Reply:
Well, for `SectionPeers`, the `self.members` shall supposed to already only contain `Joined` when insertion, so here shall no need to carry out a filter.